content/3_Silver/Prefix_Sums.mdx

```diff
@@ -307,3 +307,116 @@ for i in range(Q):
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer correct>
+      15; 17
+      <Quiz.Explanation>
+        Correct! The sum of the values of some subarray $[l, r]$ is determined by the formula $P[r] - P[l - 1]$.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      15; 21
+      <Quiz.Explanation>
+        Incorrect. Recall that to get the sum of some range of values $[l, r]$, we need to calculate $P[r] - P[l - 1]$ instead of $P[r] + P[l - 1]$. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+  </Quiz.Question>
+  <Quiz.Question>
+    **Challenge Problem**: With the idea of prefix sums in mind, at first glance it may seem logical to apply this method to return range products in constant time. Instead of adding, we multiply, and instead of subtracting, we divide. Assume that we calculate the answer modulo a given $M$ and the array has normal constraints ($N \leq 10^5$, $1 \leq a_i \leq 10^9$). What additional challenges may we face?
+    <Quiz.Answer>
+      **Integer Overflow**: Even if we use modulo, the numbers will grow too quickly when multiplying, to the point where they may not fit within a 64-bit integer.
+      <Quiz.Explanation>
+        Incorrect. We can calculate intermediate modulos, which will mitigate this problem. This property is a result of the distribute property of modulo which states that $(a \cdot b) \mod m = (a \mod m) \cdot (b \mod m_ \mod m$. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      **Time Limit**: Multiplying, dividing, and modulo are relatively slow operations, and this will make the prefix product just as fast as iterating through the array.
+      <Quiz.Explanation>
+        Incorrect. While it is true that multiplying and dividing are relatively slow, this difference is hardly enough to impact the runtime to the extent of causing TLE. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      **Memory Limit**: When we switch to products, the memory goes from $\mathcal{O}(n)$ to $\mathcal{O}(n^2)$, which will result in MLE for $n \geq 10^5$. 
+      <Quiz.Explanation>
+        Incorrect. The memory remains the same regardless of whether we use sum or product. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer correct>
+      **Division under Modulo**: We cannot directly use division as a substitute for subtraction, because division is not distributive under modulo. 
+      <Quiz.Explanation>
+        Correct! Division under modulo is not the same as addition, subtraction, or multiplication. To divide involves multiplying by the modulo inverse, which may or may not exist depending on $M$ and the value to be divided (the two should be relatively prime for there to be one). One way we could conceivably do this is if $M$ is prime itself. See [here](https://usaco.guide/gold/modular) for more details
```

Comment:
Remove the link here after moving it to the statement

Reply:
done